File: DeviceTest/src/main/java/org/lovedev/device/test/MainActivity.java
package org.lovedev.device.test;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import org.lovedev.device.test.doorscreen.DoorScreenTestActivity;
import org.lovedev.device.test.tenphone.TenTestActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void tenTest(View view) {
        startActivity(new Intent(this, TenTestActivity.class));
    }

    public void doorScreenTest(View view) {
        startActivity(new Intent(this, DoorScreenTestActivity.class));
    }
}

File: harddecode/src/main/java/com/shine/harddecode/MainActivity.kt
package com.shine.harddecode

import android.app.Activity
import android.content.Context
import android.media.AudioManager
import android.os.Bundle
import android.view.SurfaceView
import android.view.View
import shine.com.cn.testharddecode.MrVideoDecoder
import kotlin.concurrent.thread

class MainActivity : Activity() {
    lateinit var view1: SurfaceView
    private var decoder: MrVideoDecoder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        view1 = findViewById(R.id.sv_camera_1)
//        HardDecode.setSurfaceHolder(view1.holder)
    }

    fun close(view: View) {
        val audioService: AudioManager = this.getSystemService(Context.AUDIO_SERVICE) as AudioManager
        thread {
            audioService.setParameters("ForceCloseCard0Dev0")
        }
    }


    override fun onDestroy() {
//        if (decoder != null) {
//            Log.d(TAG, "onDestroy decoder release----------------")
//            decoder!!.release()
//        }

        super.onDestroy()
    }

    companion object {
        private val TAG = "testharddecode"
    }
}
